### PR TITLE
create mock test #19

### LINKED ISSUE #19: Automatically update questions on the page
We should periodically (e.g. every 5 seconds) send requests to our server and refresh our question list on the page. We should cover this functionality with a functional test.

### PR DESCRIPTION
**Pull request by vidgit-spb**
Fix #19 refresh data every 5 seconds, remake test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,10 +7,11 @@
     "build": "tsc -p src/server/tsconfig.json && tsc -p src/client/tsconfig.json",
     "start": "npm run build && node ./lib/server/index.js",
     "start-debug": "cross-env DEBUG=stackoverflowtags:* npm run start",
-    "lint": "eslint ./test ./src/server/index.js",
+    "lint": "eslint ./test/test.js ./src/server/index.js",
```
Comment (AndreyBelym) [inline package.json:10]: Please check all test files and enable [eslint-plugin-testcafe](https://www.npmjs.com/package/eslint-plugin-testcafe) if necessary.

File: src/server/defaultValues.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ dot.config();
 
 const DEFAULT_PORT   = 8000;
 const DEFAULT_ADDRESS = 'localhost';
-const maxItems = 28;
+const maxItems = 1;
```
Comment (AndreyBelym) [inline src/server/defaultValues.js:7]: Please discard this change.

File: src/server/index.js [hunk 3]
```diff
@@ -45,14 +47,24 @@ export class MyServer {
             throw err;
         }
     }
+    startServerWithMock () {
```
Comment (AndreyBelym) [inline src/server/index.js:50]: Do not use nock in the production code.

File: test/test.js [hunk 4]
```diff
@@ -1,24 +1,37 @@
 const got = require('got');
 const { expect } = require('chai');
-const server = require('../');
+const nock = require('nock');
 const { default: defaultValues } = require('../lib/server/defaultValues');
 
 describe('return type check', () => {
-    const newInstance = new server.MyServer();
+    it('Check 200 http and Name', async () => {
```
Comment (AndreyBelym) [inline test/test.js:7]: Please revert the changes in this test. You can add a new test for the `/api/questions`.

File: test/test2.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (AndreyBelym) [inline test/test2.js:1]: - Please use meaningful names (no names like `test2` and so on)
 - The best practice is storing different types of tests in different directories (`test/e2e/*`, `test/api/*`)
 - Please fix indentation and other code style issues (use only `import`, spaces after commas, newlines before multiline declarations)

File: test/testData/answer.js [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export const answer = '{"items":[{"tags":["testcafe"],"owner":{"reputation":86,"user_id":1087185,"user_type":"registered","accept_rate":27,"profile_image":"https://www.gravatar.com/avatar/aa227a526e5ac24d074373ba64729f9d?s=128&d=identicon&r=PG","display_name":"user1087185","link":"https://stackoverflow.com/users/1087185/user1087185"},"is_answered":false,"view_count":13,"answer_count":1,"score":0,"last_activity_date":1585904311,"creation_date":1585779605,"question_id":60981232,"link":"https://stackoverflow.com/questions/60981232/testcafe-how-do-you-wait-for-the-before-hook-to-complete-before-loading-the","title":"Testcafe - How do you wait for the `before` hook to complete before loading the page"}],"has_more":false,"quota_max":10000,"quota_remaining":9964}'
```
Comment (AndreyBelym) [inline test/testData/answer.js:1]: Please format the test data in a human-readable form (with indentation).

File: src/server/answer.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import qs from'qs';
```
Comment (AndreyBelym) [inline src/server/answer.js:1]: We should not keep this file.

File: src/server/api.js [hunk 8]
```diff
@@ -3,31 +3,33 @@ import qs from'qs';
...
 export default  async function getData () {
     try {    
         let bigJson = [];
         let hasMore = true;
         let maxItems = defaultValues.maxItems;
         for (let pageId = 1; hasMore && maxItems>0; pageId++) {
-            maxItems  -= params.pagesize;
-
-            const currentUrl = url.concat(`&page=${pageId}`);
+            maxItems -= objectParams.pagesize;
```
Comment (AndreyBelym) [inline src/server/api.js:27]: The previous version was better (const vs let, an additional new line)

File: test/api.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+const got = require('got');
+const { expect } = require('chai');
+const nock = require('nock');
+const { default: defaultValues } = require('../lib/server/defaultValues');
+
+describe('return type check', () => {
+    it('Check 200 http and Name', async () => {
+        const firstExpectedId = 59791234;
```
Comment (AndreyBelym) [inline test/api.js:8]: Please revert this test to the previous version. Currently you test only nock, not your application.

File: test/e2e.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        const newInstance = new server.MyServer();
+
+        scope  = await nockServer(newInstance);
+    });
+
+const TableIdSelector = Selector('#idDataTable');
+
+test('testing first data', async t => {
+
+    const firstEcpectedId = '60981232';
```
Comment (AndreyBelym) [inline test/e2e.js:24]: firstExpectedId

File: test/testData/answer.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+            'title':              'How to typetext in textarea which have dynamic element?'
+        }
+    ],
+    'has_more':        false,
+    'quota_max':       10000,
+    'quota_remaining': 9963
+};
+
+export const answer = JSON.stringify(answerParams);
+export const answer2 = JSON.stringify(answer2Params);
```
Comment (AndreyBelym) [inline test/testData/answer.js:63]: Please do not use names like answer2. originalAnswer & updatedAnswer is a lot better.

File: test/e2e.js (hunk 10 above)
Comment (vidgit-spb) [reply test/e2e.js:24]: Shame on me.

File: package.json [hunk 12]
```diff
@@ -8,9 +8,10 @@
     "start": "npm run build && node ./lib/server/index.js",
     "start-debug": "cross-env DEBUG=stackoverflowtags:* npm run start",
     "lint": "eslint ./test ./src/server/index.js",
-    "test": "npm run lint && mocha ./test/test.js",
+    "test": "npm run lint && mocha ./test/api.js",
     "watch": "npm test -- -w",
-    "lint-fix": "npm run lint -- --fix"
+    "lint-fix": "npm run lint -- --fix",
+    "testcafe": "cross-env DEBUG=stackoverflowtags:* testcafe chrome ./test/e2e.js"
```
Comment (AndreyBelym) [inline package.json:14]: Better use chrome:headless

File: package.json [hunk 13]
```diff
@@ -8,9 +8,10 @@
     "start": "npm run build && node ./lib/server/index.js",
     "start-debug": "cross-env DEBUG=stackoverflowtags:* npm run start",
     "lint": "eslint ./test ./src/server/index.js",
-    "test": "npm run lint && mocha ./test/test.js",
+    "test": "npm run lint && mocha ./test/api.js",
```
Comment (AndreyBelym) [inline package.json:11]: Better to have a separate `mocha` script & use `npm run lint && npm run mocha && npm run testcafe` here
